myshop/tests/cart/test_cart.py

```diff
@@ -0,0 +1,45 @@
+from django.test import TestCase
+from django.conf import settings
+from django.contrib.sessions.middleware import SessionMiddleware
+from django.http import HttpRequest
+from django.urls import reverse
+from cart.views import cart_add
+from tests.cart.test_model_mixin_testcase import ModelMixinTestcase
+
+
+class Testurls(ModelMixinTestcase, TestCase):
+    def test_initialize_cart_clean_session(self):
+
+        self.assertEqual(self.cart_obj.cart, {})
+
+    def test_add_new_product_to_the_cart(self):
+        self.cart_obj.add_product(self.newproduct, quantity=3)

```

Comment:
```suggestion
        self.cart.add_product(self.newproduct, quantity=3)

```
Don't add too much unnecessary detail in the name. It affects the readability